### PR TITLE
Add add schedule operation and mock test

### PR DESCRIPTION
**Pull request by wobrak**
Change NetworkService access property
Add Test for AddScheduleOperation
Add Test if App is in Mock mode
Add Fix date fix in GetScheduleResponseMapper

### CODE REVIEW

File: TOZ_iOS/GetScheduleResponseMapper.swift [hunk 1]
```diff
@@ -12,11 +12,15 @@ final class GetScheduleResponseMapper: ResponseMapper<ScheduleItem.ReservationIt
...
+            guard let dateFromJson = json["date"] as? String else { return nil }
+            let formatter = DateFormatter()
+            formatter.dateFormat = "yyyy-MM-dd"
+            formatter.timeZone = TimeZone(abbreviation: "UTC")
+            guard let date = formatter.date(from: dateFromJson) else { return nil }
             guard let startTime = json["startTime"] as? String else { return nil }
             guard let endTime = json["endTime"] as? String else { return nil }
             var timeOfDay: TimeOfDay
-            if startTime == "08:00" || endTime == "12:00" {
+            if startTime == "08:00" && endTime == "12:00" {
```
Comment (piotr-tobolski) [inline TOZ_iOS/GetScheduleResponseMapper.swift]: I don't know what errors this was producing but I would just not use the `endTime` and we will have stonger guarantee that nothing falls into wrong category.

File: TOZ_iOS/ServiceOperation.swift [hunk 2]
```diff
@@ -12,7 +12,7 @@ Class creates BackendService instance internally so it doesn’t need to be crea
  */
 class ServiceOperation {
 
-    let service: BackendService
+    var service: BackendService
```
Comment (piotr-tobolski) [inline TOZ_iOS/ServiceOperation.swift]: Why this is `var`?

File: TOZ_iOSTests/AMockIfSetTest.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+//  AMockIfSetTest.swift
+//  TOZ_iOS
+//
+//  Copyright © 2017 intive. All rights reserved.
+//
+
+import XCTest
+@testable import TOZ_iOS
+
+class AMockIfSetTest: XCTestCase {
```
Comment (piotr-tobolski) [inline TOZ_iOSTests/AMockIfSetTest.swift]: This name is just wrong.

`ServiceOperationTests`

But... You are actually testing one od it's dependencies so please change the logic a bit and call it

`BackendServiceTests`

File: TOZ_iOSTests/AMockIfSetTest.swift [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    var serviceOperation: ServiceOperation?
+
+    override func setUp() {
+        serviceOperation = ServiceOperation()
+    }
+
+    func testAddScheduleResponse() {
```
Comment (piotr-tobolski) [inline TOZ_iOSTests/AMockIfSetTest.swift]: You are definitely not testing the `AddScheduleResponse` there

File: TOZ_iOSTests/AddScheduleResponseMapperTests.swift [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+//  AddScheduleResponseMapperTests.swift
+//  TOZ_iOS
+//
+//  Copyright © 2017 intive. All rights reserved.
+//
+
+import XCTest
+@testable import TOZ_iOS
+
+class AddScheduleResponseMapperTests: XCTestCase {
```
Comment (piotr-tobolski) [inline TOZ_iOSTests/AddScheduleResponseMapperTests.swift]: You are testing also `AddScheduleOperation` there. Please separate those tests or test just the `AddScheduleResponseMapper`.

File: TOZ_iOSTests/AddScheduleResponseMapperTests.swift [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+    func checkAddScheduleResponse() {
+        XCTAssertEqual(self.scheduleItem?.idObject, "111111-347f-4b2e-b1c6-6faff971f767")
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd"
+        formatter.timeZone = TimeZone(abbreviation: "UTC")
+        XCTAssertEqual(formatter.string(from: (self.scheduleItem?.date)!), "2017-01-21")
+        XCTAssertEqual(self.scheduleItem?.timeOfDay, TimeOfDay.afternoon)
+        XCTAssertEqual(self.scheduleItem?.ownerForename, "John")
+        XCTAssertEqual(self.scheduleItem?.ownerSurname, "Doe")
```
Comment (piotr-tobolski) [inline TOZ_iOSTests/AddScheduleResponseMapperTests.swift]: Asserts should be in different test methods

File: TOZ_iOSTests/AddScheduleResponseMapperTests.swift (hunk 5 above)
Comment (piotr-tobolski) [inline TOZ_iOSTests/AddScheduleResponseMapperTests.swift]: what about this problem? Now I see more issues. This may fail at some random place, you have to specifically also wait for async operations: `waitForExpectationsWithTimeout:handler:` I suggest to remove the operation from those tests and test just the mapper.

File: TOZ_iOSTests/BackendServiceTests.swift [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+//
+//  Copyright © 2017 intive. All rights reserved.
+//
+
+import XCTest
+@testable import TOZ_iOS
+
+class BackendServiceTests: XCTestCase {
+
+    var serviceOperation: ServiceOperation?
```
Comment (piotr-tobolski) [inline TOZ_iOSTests/BackendServiceTests.swift]: ???
You should be testing just the BackendService, not the ServiceOperation there...

Comment (wobrak) [inline TOZ_iOSTests/BackendServiceTests.swift]: removed . it will be in another PR

Comment (wobrak) [inline TOZ_iOSTests/BackendServiceTests.swift]: only mapper test remained
